test/Conversion/VectorToAIEVec/test_mac_elem.mlir

```diff
@@ -0,0 +1,217 @@
+// RUN: aie-opt %s --convert-vector-to-aievec="aie-target=aieml" | FileCheck %s
+
+module {
```

Comment:
IMO this example should be cut down. For example you could change the code so that only 2 products are summed in the inner loop. The transform should apply equally.

`
module {
  func.func @matmul(%arg0: memref<64x64xi32>, %arg1: memref<64x64xi32>, %arg2: memref<64x64xi32>) {
    %c0_i32 = arith.constant 0 : i32
    affine.for %arg3 = 0 to 64 {
      affine.for %arg4 = 0 to 64 step 16 {
        affine.for %arg5 = 0 to 64 step 16 {
          %0 = vector.transfer_read %arg0[%arg3, %arg5], %c0_i32 {in_bounds = [true], permutation_map = affine_map<(d0, d1) -> (0)>} : memref<64x64xi32>, vector<16xi32>
          %1 = vector.transfer_read %arg1[%arg5, %arg4], %c0_i32 {in_bounds = [true]} : memref<64x64xi32>, vector<16xi32>
          %2 = arith.muli %0, %1 : vector<16xi32>
          %3 = vector.transfer_read %arg2[%arg3, %arg4], %c0_i32 {in_bounds = [true]} : memref<64x64xi32>, vector<16xi32>
          %4 = arith.addi %3, %2 : vector<16xi32>
          %5 = affine.apply affine_map<(d0) -> (d0 + 1)>(%arg5)
          %6 = vector.transfer_read %arg0[%arg3, %5], %c0_i32 {in_bounds = [true], permutation_map = affine_map<(d0, d1) -> (0)>} : memref<64x64xi32>, vector<16xi32>
          %7 = affine.apply affine_map<(d0) -> (d0 + 1)>(%arg5)
          %8 = vector.transfer_read %arg1[%7, %arg4], %c0_i32 {in_bounds = [true]} : memref<64x64xi32>, vector<16xi32>
          %9 = arith.muli %6, %8 : vector<16xi32>
          %10 = arith.addi %4, %9 : vector<16xi32>
          vector.transfer_write %94, %arg2[%arg3, %arg4] {in_bounds = [true]} : vector<16xi32>, memref<64x64xi32>
        }
      }
    }
    return
  }
}
`